Improvement: Ensure respect for --no-deprecation flag when adding non…

EugeneHlushko opened the pull request:
… util.depracation warnings to console output

<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->

**What kind of change does this PR introduce?**
Account for `--no-deprecation` flag while pushing deprecation warnings outside of `util.deprecate`

<!-- E.g. a bugfix, feature, refactoring, build related change, etc… -->

**Did you add tests for your changes?**
No

<!-- Note that we won't merge your changes if you don't add tests -->

**If relevant, link to documentation update:**

<!-- Link PR from webpack/webpack.js.org here, or N/A -->

**Summary**
After looking at #5942 i thought that adding a flag to hide 2-3 deprecation warnings in each place wasnt the best idea, if we really dont want to see the messages which we actually trigger with tests, we should be able to change the `test:unit` and `test:integration` commands to use `--no-deprecation` instead of doing `--trace-deprecation` and per call change to `process.noDeprecation` value.

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**
No

<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

**Other information**
None

CC: @madeinfree

sokra commented:
The test for the `apply` method is [here](https://github.com/webpack/webpack/blob/2b1046e0a9e99c95d6cc5a3043d3c6f0967c5b1d/test/Compiler.test.js#L180-L189). You could add a test for the `plugin` method too to get full test coverage.

EugeneHlushko commented:
Hi @sokra 
I've added a test for `plugin` and moved to `util.deprecation`, but now as i dont push to warnings the test becomes redundant, any other way that we can check it here: https://github.com/webpack/webpack/pull/5985/files#diff-8f9ae416209a5b226aebb96b28bb7fb9R97

sokra commented:
Ok just leave the NoErrorsPlugin as it was before the changes. It's doesn't log to stdout so I guess it's not relevant. I'll remove the plugin in the next branch, as it's deprecated since v2 anyway.

EugeneHlushko commented:
@sokra think its good now, i've seen you already made removal #5988 👍

sokra commented:
Thanks

sokra commented on lib/Compiler.js:
```diff
@@ -186,35 +187,30 @@ class Compiler extends Tapable {
+					const args = arguments;
+					this.plugin("compilation", (compilation, data) => {
+						data.normalModuleFactory.plugin("parser", parser => {
+							parser.apply(parser, args);
```
No no a bug sneeked into this:

It should be `parser.apply.apply(parser, args);` instead of `parser.apply(parser, args);`

Could you send a new PR fixing this?

madeinfree commented:
Thank you @EugeneHlushko, @sokra could I PR to help fix this 💪?

madeinfree commented:
@sokra OK~ I send a new PR here https://github.com/webpack/webpack/pull/6005, thank you !!
